plugins/uefi/fu-plugin-uefi.c

```diff
@@ -768,11 +777,14 @@ fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 			continue;
 		}
 		fu_device_set_quirks (FU_DEVICE (dev), fu_plugin_get_quirks (plugin));
-		fu_uefi_device_set_esp (FU_UEFI_DEVICE (dev), data->esp);
+		if (data->esp != NULL)
+			fu_uefi_device_set_esp (FU_UEFI_DEVICE (dev), data->esp);
 		if (!fu_plugin_uefi_coldplug_device (plugin, dev, error))
 			return FALSE;
 		if (error_efivarfs != NULL) {
 			fu_device_set_update_error (FU_DEVICE (dev), error_efivarfs->message);
+		} else if (error_udisks2 != NULL) {
+			fu_device_set_update_error (FU_DEVICE (dev), error_udisks2->message);
```

Comment:
Not a criticism of this patch at all, but I do wonder if this is getting to the point where we should have actual API for a plugin to return a message to the user rather than either creating fake devices or setting the update error. It would certainly clean up a few plugins considerably.

e.g. an API like GetPlugins() where we return an array of `[ boolean:enabled, string:summary, string:error_message ]` which we use before we show the device tree in `fwupdmgr get-devices` or `fwupdmgr get-updates`.

Reply:
Up until now we have been pretty prescriptive to not expose to the client that the implementation in the daemon is with plugins and they get disabled. So part of the problem can be older clients don't know how to show this information.

So if we're going to do such a thing maybe it's time to make a harder line in sand on versioning.

Reply:
One more thought - there are a lot of plugins that get disabled and it's fully intentional. For example if you don't have a Dell system you won't have dell plugin, and if you have a uefi system you won't have coreboot.

Reply:
> So part of the problem can be older clients don't know how to show this information

Valid. Maybe in this case, server side, we can set the FuDevice update_error IF the device is not UPDATABLE and the FuPlugin has an error set?

> there are a lot of plugins that get disabled and it's fully intentional

I guess we either don't set an update message in that case (i.e. it's working as desired) or we make the `boolean:enabled` into an enum like _DISABLED_NOTIFY, _DISABLED_INTENTIONAL, ENABLED, etc.

Reply:
Can we move the discussion to a new issue to figure out the solution? I think that this commit should backport directly into 1_4_X as is, and we can reswizzle for 1.5.1?